rotkehlchen/chain/evm/transactions.py

```diff
@@ -456,19 +457,52 @@ def get_or_query_transaction_receipt(self, tx_hash: EVMTxHash) -> 'EvmTxReceipt'
             )
 
         if len(result) == 0:
-            transaction = self.evm_inquirer.get_transaction_by_hash(tx_hash)
-            with self.database.user_write() as write_cursor:
-                dbevmtx.add_evm_transactions(write_cursor, [transaction], relevant_address=None)
-            period = TimestampOrBlockRange(
-                range_type='blocks',
-                from_value=transaction.block_number,
-                to_value=transaction.block_number,
-            )
-            if transaction.to_address is not None:  # internal transactions only through contracts
-                self._query_and_save_internal_transactions_for_range(
-                    address=None,  # get all internal transactions for the parent hash
-                    period=period,
+            if tx_hash == GENESIS_HASH:
+                # For each tracked account, query to see if it had any transactions in the
+                # genesis block
+                with self.database.conn.read_ctx() as cursor:
+                    accounts_data = self.database.get_blockchain_account_data(
+                        cursor=cursor,
+                        blockchain=self.evm_inquirer.chain_id.to_blockchain(),
+                    )
+                    for data in accounts_data:
+                        queried_range = self.database.get_used_query_range(
+                            cursor=cursor,
+                            name=self.evm_inquirer.chain_id.name + 'txs_' + data.address,
```

Comment:
This is wrong. Notice how it's done in the rest of this file

```python
        location_string = f'{self.evm_inquirer.blockchain.to_range_prefix("txs")}_{address}'
```

The result is not the same. Which also means this may not be well tested. Can you add a test here that will fail with your current implementation and reveal the mistake?

Reply:
Fixed